eth/backend.go

```diff
@@ -567,6 +577,9 @@ func (s *Ethereum) Stop() error {
 		s.lesServer.Stop()
 	}
 
+	if err := s.chainDb.Delete([]byte("unsafe-shutdown")); err != nil {
+		log.Error("err", err)
+	}
```

Comment:
You'll need to move this down, right above `chaindb.Close()`. The biggest offender for an unclean shutdown is the blockchain shutdown mechanism.